src/crystal/project/git_dependency.cr

```diff
@@ -0,0 +1,60 @@
+require "crystal/project/dependency"
+require "crystal/project/project_error"
+
+module Crystal
+  class GitDependency < Dependency
+    getter target_dir
+
+    def initialize(@repo, name = nil : String?, branch = nil : String?)
+      unless repo =~ /(.*)(:|\/)(.*)\/(.*)(.git)/
```

Comment:
I agree, I think GitDependency should have "name" as required, we shouldn't infer anything from the repository url.